src/main/java/appeng/core/sync/packets/PacketCompressedNBT.java

```diff
@@ -28,93 +28,108 @@
 import java.util.zip.GZIPOutputStream;
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.Unpooled;
 
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiScreen;
-import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.nbt.CompressedStreamTools;
 import net.minecraft.nbt.NBTTagCompound;
 
-import cpw.mods.fml.relauncher.Side;
-import cpw.mods.fml.relauncher.SideOnly;
+import cpw.mods.fml.common.network.simpleimpl.MessageContext;
 
 import appeng.client.gui.implementations.GuiInterfaceTerminal;
 import appeng.core.sync.AppEngPacket;
-import appeng.core.sync.network.INetworkInfo;
+import appeng.core.sync.AppEngPacketHandler;
 
 
-public class PacketCompressedNBT extends AppEngPacket
+public class PacketCompressedNBT implements AppEngPacket, AppEngPacketHandler<PacketCompressedNBT, AppEngPacket>
 {
 
 	// input.
-	final NBTTagCompound in;
+	private NBTTagCompound in;
 	// output...
-	private final ByteBuf data;
-	private final GZIPOutputStream compressFrame;
-	int writtenBytes = 0;
-	boolean empty = true;
+	private ByteBuf data;
+	private GZIPOutputStream compressFrame;
+	private int writtenBytes = 0;
+	private boolean empty = true;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "empty" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)